src/AlgebraicGeometry/Schemes/AffineSchemes/Objects/Properties.jl

```diff
@@ -654,6 +650,53 @@ end
 is_smooth(X::AbsSpec{<:Field, <:MPolyRing}) = true
 is_smooth(X::AbsSpec{<:Field, <:MPolyLocRing}) = true
 
+###################################################################
+# Irreducibility and Integrality                                  #
+#    integral iff (irreducible && reduced)                        #
+#    integral = OO(X) is integral domain                          #
+#    irreducible = nilradical of OO(X) is prime                   #
+###################################################################
+@doc Markdown.doc"""
+   is_irreducible(X::AbsSpec)
+
+Return the boolean value whether an affine scheme `X` is irreducible.
+
+"""
+@attr function is_irreducible(X::AbsSpec)
+  !is_empty(X) || return false
+  !get_attribute(X, :is_integral, false) || return true 
+                                           ## integral = irreducible + reduced
+  return (length(minimal_primes(saturated_ideal(modulus(OO(X))))) == 1)
```

Comment:
What if the base ring is not an integral domain? or a field? is this correct? throw an error? 
Looks to me like we should be able to ask a ring whether it is an integral domain. 
Then the ring should know best what to do.

Reply:
We have to look at two different settings:
A) For the given ring OO(X) minimal_primes/primary_decomposition/is_integral_domein is implemented, then the problem is solved on the algebraic side, i.e. we obtain a correct answer for the given ring. In particular, the answer will be over the given field k for k[x_1,..,x_n]/I and not over the algebraic closure, but this is the way it should be for scheme.
B) For the given ring OO(X) these functions are not available, then the algebra side will throw an error.
But I agree that I should put a comment in the documentation pointing out the subtlety of the underlying field.

Reply:
- now restricted to schemes of MPolyAnyRing
- docstring added